File: app/src/main/kotlin/com/openclaw/ghostcrab/domain/model/OpenClawConfig.kt
package com.openclaw.ghostcrab.domain.model

import kotlinx.serialization.json.JsonElement

/**
 * The full `openclaw.json` configuration as returned by the gateway.
 *
 * Sections are kept as raw [JsonElement] to avoid brittle schema lock-in.
 * Phase 6 adds typed wrappers for well-known sections (`gateway.http`, `gateway.auth`, etc.)
 * while this class remains the authoritative wire type.
 *
 * @param sections Top-level config keys mapped to their raw JSON values.
 * @param etag Server-side version tag for optimistic-concurrency detection. May be `null` if
 *   the gateway does not support ETags.
 */
data class OpenClawConfig(
    val sections: Map<String, JsonElement>,
    val etag: String? = null,
)

File: ios/GhostCrab/Domain/Model/OpenClawConfig.swift
import Foundation

/// The full `openclaw.json` configuration as returned by the gateway.
///
/// Sections are kept as raw `AnyCodable` JSON values to avoid brittle schema lock-in.
/// Phase 6 adds typed wrappers for well-known sections (`gateway.http`, `gateway.auth`, etc.)
/// while this type remains the authoritative wire type.
///
/// - Parameters:
///   - sections: Top-level config keys mapped to their raw JSON values.
///   - etag: Server-side version tag for optimistic-concurrency detection. May be `nil`
///     if the gateway does not support ETags.
public struct OpenClawConfig: Sendable {
    public let sections: [String: AnyCodable]
    public let etag: String?

    public init(sections: [String: AnyCodable], etag: String? = nil) {
        self.sections = sections
        self.etag = etag
    }
}
